docs/developer-guide.md

```diff
@@ -159,6 +159,14 @@ actionHandler.mockReturnValueOnce({
 });
 ```
 
+To maintain test isolation, we should reset (note that this is different from clear) mocks before every test:
+
+````
+beforeEach(() => {
+    vi.resetAllMocks();
+
+  });```
+
```

Comment:
Thanks for this, I think this is a great idea and have refactored some of my tests. However I'm not sure if we need to add it before every test or just tests that we use mocks for? 

I don't feel strongly either way, do you think it's useful to do before even simple tests for example some of the util tests?

Reply:
I think it would only possibly affect tests with functions/components that use those imports.

Another file I looked at refactoring is this [actions.test.ts](https://github.com/bcgov/cas-registration/blob/develop/client/tests/utils/actions.test.ts) test. The global `consoleMock` and `sentryMock` that we call throughout the tests will break if run `resetAllMocks()` and we can't put them in the `beforeEach` since they are called through the test. 

Anyways I think this is a great best practice that we might have to deviate from in certain tests like the actions test unless you have any refactoring ideas.

Reply:
I'm going to update these docs to talk about reset vs clear because I think there are cases, like actions, where one is preferred over the other. I found this in the actions file:

```
// disable the global mock since we are testing actions here
vi.unmock("@/app/utils/actions");
``` so it looks like tests could be using mocks without explicit imports (?), so I'll mention that too.